core/src/main/java/org/apache/iceberg/rest/RESTSessionCatalog.java

```diff
@@ -367,46 +368,67 @@ public boolean tableExists(SessionContext context, TableIdentifier identifier) {
   }
 
   private LoadTableResponse loadInternal(
-      SessionContext context, TableIdentifier identifier, SnapshotMode mode) {
+      SessionContext context,
+      TableIdentifier identifier,
+      SnapshotMode mode,
+      Map<String, Object> viewContext) {
     Endpoint.check(endpoints, Endpoint.V1_LOAD_TABLE);
     AuthSession contextualSession = authManager.contextualSession(context, catalogAuth);
+
+    // inject the context to query params
+    Map<String, String> queryParams = Maps.newHashMap(mode.params());
+
+    Object viewIdentifierObj = viewContext.get(ContextAwareTableCatalog.VIEW_IDENTIFIER_KEY);
+    if (viewIdentifierObj instanceof String) {
+      String viewIdentifierString = (String) viewIdentifierObj;
+      String[] parts = viewIdentifierString.split("\\.");
+      if (parts.length >= 2) {
+        String[] namespaceParts = new String[parts.length - 1];
+        System.arraycopy(parts, 0, namespaceParts, 0, parts.length - 1);
+        String viewName = parts[parts.length - 1];
+        queryParams.put(
```

Comment:
Just for my understanding:
Let's take the following input: "ns1.ns2.view1"
We convert this to "ns1%1Fns2%1F.view1"
Is this because we wan't to tell the REST catalog which part is the namespace and which is the view name? If we include a dot in the query params, can't we just send the view path as it is with dots and let the catalog server figure out the namespace/view name part that we do here? Or can we also use the same special char between the namespace and the view name?

Reply:
one way to look at this is anything from the last `.` to the end is the view name and the every thing before it is the namespace, the reason `%1F` was chosen the separator was to use something similar, the reason i choose the `%1F` is the language of the existing rest spec for the nested namespaces.

I think we are yet to conclude on the separator discussion.